### PR TITLE
Tag version note

### PR DESCRIPTION
**Pull request by zancas**
In this PR add notes about:

*   the possibility of needing to start an ssh-agent
*   how to choose a tag to build against, with a preference for the most recent `rc` tag
*   move verification paragraph above instructions about opening a PR

### CODE REVIEW

**Comment by garethtdavies** [comment]
> * how to choose a tag to build against, with a preference for the most recent `rc` tag

Curious why target the most recent `rc` tag and simply not the latest release?

**Comment by zancas** [comment]
No good reason.  I'll fix this.

**Comment by zancas** [comment]
I had thought, that deterministic builds should be demonstrated prior to a release (hence while the release is still a candidate).

In retrospect, I don't understand the release process well enough to know this is correct.

If the accepted release candidate is (as I had assumed) the exact code used in the release, and if there aren't so many releases that gitian builders would be overwhelmed, then I think it makes good sense to have multiple attestations of the `rc` _before_ it becomes the release.

**Comment by zancas** [comment]
Now that I think about it, and because I don't know how minor bug-fixes, hotfixes, are handled, my grep might grab the wrong thing.

Basically, I wanted to "gitian attest" to the latest thing in production, and assumed that others would want to do that as well.

(I'm grateful for feedback/clarification.)

**Comment by charlieok** [comment]
I think you are both right. There is some value in comparing results...
a) on any commit (to verify determinism in the build process)
and additional value for
b) releases (to help build assurance in the integrity of a copy of the release).

(a) is possible to automate, and we want to add that to our CI system. That should make all of this easier in the future.
(b) is in some measure about trust, and can never really be solved by us, the publishers, by ourselves. This is what the signatures actually pushed and committed to the gitian.sigs repository are about.

**Comment by charlieok** [comment]
In the docs in question, I didn't go into how or for what reasons someone might choose a particular commit or tag to build. Maybe it would be helpful to include some of the stuff discussed in this thread to help clarify that. As gareth notes, giving special preference to 'rc' tags doesn't seem like the answer.

**Comment by zancas** [comment]
> In the docs in question, I didn't go into how or for what reasons someone might choose a particular commit or tag to build. Maybe it would be helpful to include some of the stuff discussed in this thread to help clarify that. As gareth notes, giving special preference to 'rc' tags doesn't seem like the answer.

@garethtdavies has explained why targeting `rc` tags for attestation is wrong.  I will fix that.

File: README.md [hunk 1]
```diff
@@ -477,9 +477,13 @@ zcash-gitian$ vagrant plugin install --local
 ## Configure the version of zcash you want to build and sign
 
 Set the value of the `ZCASH_VERSION` variable in `.env` to point to the zcash commit you want to
-create a signature for. Likely you want the name of a git tag, such as `v2.0.5-2`.
-
+create a signature for. Likely you want the name of a git tag, usually the most recent released
+version.  You can determine this by running the following command in a freshly updated clone of the
+zcash repository:
```
Comment (zancas) [inline README.md]: @garethtdavies @charlieok how's this?   Specifically _excluding_ `rc`s?

Comment (charlieok) [reply README.md]: Running that I get:
```
zcash$ git tag | grep -v rc |  tail -n 1
zc.v0.11.2.z0
```

We'll want the tags ordered by version number, and we'll want to exclude tags that don't match the pattern we're looking for. How about something like:

```
zcash$ git tag --sort=version:refname | grep ^v | grep -v rc |  tail -n 1
v2.0.6
```

Comment (zancas) [reply README.md]: Huh!  What version of `git` are you using?

Mine:

```
git --version
git version 2.22.0
```

Comment (charlieok) [reply README.md]: $ git --version
git version 2.21.0 (Apple Git-120)

Seems unlikely to be due to a difference in git version. Does my command work on your system?

Comment (garethtdavies) [reply README.md]: This command works but seems a bit overkill. Why not link to Github tags page https://github.com/zcash/zcash/tags?

Comment (charlieok) [reply README.md]: Sounds like a good idea to me

Comment (zancas) [reply README.md]: > ```
> $ git --version
> git version 2.21.0 (Apple Git-120)
> ```
> 
> Seems unlikely that these commands would work differently on our two versions of git. Is that what you're seeing? Does the command I suggested give the desired output on your setup?

I tested your command, and it gave the desired output.  I also agree that it seems implausible that `git tag`s interface would change (just wanted to rule it out).

Comment (zancas) [reply README.md]: > This command works but seems a bit overkill. Why not link to Github tags page https://github.com/zcash/zcash/tags?

Acknowledged, I will make this change.

**Pull request merged**